app/src/server/trpc/routers/service.ts

```diff
@@ -36,6 +37,25 @@ export const serviceRouter = router({
       return artifactList;
     }),
 
+  getDataRequirements: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']), id: z.string() }))
+    .query(async ({ input }) => {
+      const res = await fetch(
+        `${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}/${input.id}/$data-requirements`
+      );
+      const resource = await res.json();
+      if (resource?.resourceType === 'OperationOutcome') {
+        const eMessage = resource?.issue[0]?.details?.text;
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
```

Comment:
So you can also make the `code` here flexible since it is something that would be on the operation outcome. Try accessing it like you access the message on `resource?.issue[0]?.details?.text`!

Reply:
I'm a bit confused, do you want me to directly set the message variable to resource?.issue[0]?.details?.text or something else?

Reply:
I'm suggesting that rather than hardcoding code to be 'INTERNAL_SERVER_ERROR', you can have it be equal to whatever `code` is on the OperationOutcome. So, `code: resource?.issue[0]?.code`

Reply:
ohhh, okay thank you!